front-end/src/hooks/ApiHooks.ts

```diff
@@ -29,4 +29,17 @@ const useWebSocket = () => {
   return {data}
 }
 
-export default useWebSocket
+export const getDevices = async () => {
+  const [devices, setDevices] = useState([])
+
+  useEffect(() => {
+    const fectDevices = async () => {
+      const response = await fetch(apiUrl)
+      const data = await response.json()
+      setDevices(data)
+    }
+
+    fectDevices()
+  }, [])
+  return devices
+}
```

Comment:
Modify it to fit the reacts hook rules. We can later on add more device related api calls into the same hook, and then return them, as needed, in the destructured return statement. This also requires you to change the function call in the app.ts file
```suggestion
export const useDevices = async () => {
  const [devices, setDevices] = useState([])

  useEffect(() => {
       fetchDevices ()
  }, [])
  const fetchDevices = async () => {
      const response = await fetch(apiUrl)
      const data = await response.json()
      setDevices(data)
    }
  return {devices}
}
```

Reply:
Did as suggested.